File: ch06/42_1/ReverseSentence.cpp
class Solution {
public:
    //用两步翻转法，第一步先把整个句子逆序翻转，第二步再翻转每个单词中的顺序。
    string ReverseSentence(string str) {
        
        if( str.length() == 0)
            return str;
        
        reverse(str.begin(), str.end());//翻转整个字符串
        
        int begin = 0, end = 0;
        while(end <  (int)str.length()){
            
            if(str[begin] == ' '){
                begin ++;
                end ++;
            }
       		while(end < str.length() && str[end] != ' '){
                end ++;
            }
            reverse(str.begin() + begin, str.begin() + end);
            begin = ++end;
            
        }
        return str;
       
    }
};

File: ch06/42_1/ReverseSentence.py
# -*- coding:utf-8 -*-
class Solution:
    def ReverseSentence(self, s):
        # write code here
        s_lst = s.split(" ")
        for i in range(len(s_lst)):
            s_lst[i] = s_lst[i][::-1]
        
        return "".join(s_lst)[::-1]
